lib/scrive/eid.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+require 'uri'
+require 'net/http'
+require 'json'
+require 'pry'
+
+module Scrive
+  # Client to connect to eID API
+  class EID
+    class TransactionNotCompletedError < StandardError; end
+
+    class RateLimitError < StandardError; end
+
+    class BadRequestError < StandardError; end
+
+    attr_reader :token, :base_uri
+
+    def initialize(base_uri:, token: nil)
+      @token = token
+      @base_uri = base_uri
+    end
+
+    def new_transaction(redirect_url:)
+      url = URI("#{base_uri}/transaction/new")
+
+      request = Net::HTTP::Post.new(url)
+      request['Authorization'] = authorization_header
+      request.body = build_request_body(redirect_url)
+
+      response = http(url).request(request)
+      parse_response(response)
+    end
+
+    def verify_transaction?(transaction_id:, ssn:)
+      response = get_transacation(transaction_id: transaction_id)
+      raise TransactionNotCompletedError if response['status'] != 'complete'
+
+      response.dig(*ssn_path) == ssn.to_s
+    end
+
+    private
+
+    def ssn_path
+      %w[providerInfo seBankID completionData user personalNumber]
+    end
+
+    def http(url)
+      Net::HTTP.new(url.host, url.port).tap do |http|
+        http.use_ssl = true
+        http.verify_mode = OpenSSL::SSL::VERIFY_PEER
+      end
+    end
+
+    def get_transacation(transaction_id:)
```

Comment:
Hur används denna här? `autoStartToken` eller är det den man indirekt skickar in som transaction ID, tänker på deras dokumentation, se nedan.

```Swedish BankID
Create a transaction.
Start the transaction, extract autoStartToken from the response, use it in custom UI for end user. More information could be found in BankID Relying Party Guidelines on BankID documentation. (See Chapter 3, "Launching". Please note that you'd need to provide redirect parameter and that it's not related to redirectUrl one provides on transaction creation.)
Poll transaction status and completion data.
Please take into account that SE BankID doesn't allow more than one ongoing transactions for a given personal number. Therefore if you specify end user's personal number on creation or starting of the transaction, you need to cancel it before being able to start a new transaction for the same personal number again. (Or wait until the transaction has expired, the expiration time is fixed by SE BankID and is 3 minutes.)```

Reply:
Vi använder inte den endpointen utan går via Scrives frontend vilket de rekommenderar. Då behöver vi inte polla heller

https://testbed-eid.scrive.com/documentation/api/v1/#null%2Fpaths%2F~1transaction~1new%2Fpost